packages/modules-sdk/src/medusa-module.ts

```diff
@@ -0,0 +1,108 @@
+import { asValue } from "awilix"
+import { createMedusaContainer } from "medusa-core-utils"
+import { moduleLoader, registerMedusaModule } from "./loaders"
+import {
+  ExternalModuleDeclaration,
+  InternalModuleDeclaration,
+  MODULE_RESOURCE_TYPE,
+  MODULE_SCOPE,
+} from "./types"
+import { loadModuleMigrations } from "./loaders/utils"
+
+const logger: any = {
+  log: (a) => console.log(a),
+  info: (a) => console.log(a),
+  warn: (a) => console.warn(a),
+  error: (a) => console.error(a),
+}
+
+export class MedusaModule {
+  public static async bootstrap(
+    moduleKey: string,
+    defaultPath: string,
+    declaration?: InternalModuleDeclaration | ExternalModuleDeclaration,
+    injectedDependencies?: Record<string, any>
+  ): Promise<{
+    [key: string]: any
+  }> {
+    let modDeclaration = declaration
+    if (declaration?.scope !== MODULE_SCOPE.EXTERNAL) {
+      modDeclaration = {
+        scope: MODULE_SCOPE.INTERNAL,
+        resources: MODULE_RESOURCE_TYPE.ISOLATED,
+        resolve: defaultPath,
+        options: declaration,
+      }
+    }
+
+    const container = createMedusaContainer()
+
+    if (injectedDependencies) {
+      for (const service in injectedDependencies) {
+        container.register(service, asValue(injectedDependencies[service]))
+      }
+    }
+
+    const moduleResolutions = registerMedusaModule(moduleKey, modDeclaration!)
+
+    await moduleLoader({ container, moduleResolutions, logger })
+
+    const services = {}
+
+    for (const resolution of Object.values(moduleResolutions)) {
+      const containerName = resolution.definition.registrationName
```

Comment:
suggestion: ahah I come back again, but can we rename `containerName` to just `registrationName`